Feature/mail incident status

mkcorpc opened the pull request:
Please find enclosed the implemented responder.

[FR] Responder which sends a mail with a detailed incident status #920

ntcong commented on responders/MailIncidentStatus/MailIncidentStatus.json:
```diff
@@ -0,0 +1,98 @@
+      "type": "string",
+      "multi": true,
+      "required": false,
+      "defaultValue": "Incident Case Notification: "
```
this should be ["Incident Case Notification: "] or better ["address@example.com"]

mkcorpc replied:
Yes correct. 

Sorry it is my first pull request. Do I need to do something to confirm this change?

ntcong replied:
I don't have reviewer status so it's just a comment

I tried your plugin on thehive4 and it displays a list of separate character instead of a string

mkcorpc replied:
> I tried your plugin on thehive4 and it displays a list of separate character instead of a string

I dont know what your a meaning with 'separate characters'. Do you speak about an configuration option?

The following mail is sent by the plugin:
![mail](https://user-images.githubusercontent.com/23398789/103939351-ec031280-512b-11eb-8ba7-12de42868be5.png)

ntcong replied:
![image](https://user-images.githubusercontent.com/167336/104019034-68115f00-51ed-11eb-9433-63005c83a5a6.png)

it looks like this from the settings page

dadokkio replied:
I can confirm ntcong error and suggestion.
First line is with his fix `"defaultValue": ["Incident Case Notification:"]`
second whith standard code "`defaultValue": "Incident Case Notification: "`
![image](https://user-images.githubusercontent.com/355183/104021565-48fbd880-51bf-11eb-8d2a-30a5a056defd.png)

dadokkio commented:
I'm going to test the responder now. 
Quick suggestion, can you remove ssl, email and json from requirements. Those are standards libs and trying to install requirements with pip it tries to install wrong ssl lib instead.

dadokkio commented:
Some issues during tests:

- My case don't have data.updatedAt value set so it fails when tries to divide it by 1e3
- I've seen that you have commented out server.send_message(msg) .. it should be ok since you put From and To in the msg so are not required as parameters. Also [mail_address] returns an error since a list is not a valid option
- I've custom fields that are not set in this case, but the create_HTMLTable function returns error because i[1] is then None and cannot be concatenate
- When you check for email in tags can you please check for both "mail=" and "mail:"? TH3 and TH4 have different behavior on tags management and you need to check both.

mkcorpc commented:
I have implemented the following changes:
- [X] Removed ssl, email and json from requirements
- [X] Removed the comment 'server.send_message(msg)'
- [ ] [mail_address] returns an error since a list is not a valid option
- [ ]

mkcorpc commented:
Addjusted json configuration. 
- [X] Deleted default value for tlp_amber_mail_addresses
- [X] Deleted default value for tlp_green_mail_domains

dadokkio commented:
I was able to run the plugin :smile:, I just put small comments in the code.
The multilined ().format syntax for text I proposed for the summary could be also used in email creation.

mkcorpc commented:
Sorry, but I don't know where i can find your comments. How can I see these?

ntcong commented on responders/MailIncidentStatus/mailincidentstatus.py:
```diff
@@ -0,0 +1,272 @@
+        html += "<table width=\"100%\" cellpadding=\"0\" cellspacing=\"0\" border=\"0\" bgcolor=\"#FFFFFF\" align=\"center\">\n"
+        html += "<colgroup><col class=\"first\"/><col class=\"second\"/></colgroup>\n"
+        html += "<tr><td colspan=\"2\">Incident Status Report</td></tr>\n"
+        for i in two_tuple_list:
```
updatedBy can be None: ('updatedBy', None) which will trigger error:
`html += i[1] TypeError: can only concatenate str (not "NoneType") to str`

dadokkio commented:
I updated your code with my and @ntcong suggestions. Let me know your opinion :+1: 
It seems all changed cause I've black formatting on save :smile:

mkcorpc commented:
I have "accepted" all conversations.
